src/Umbraco.Web.UI.Login/src/components/external-login-provider.element.ts

```diff
@@ -0,0 +1,206 @@
+	 * Setting this property will cause the default view to be hidden and the custom view to be loaded.
+	 * The icon, button look and button color will be ignored.
+	 *
+	 * @example App_Plugins/MyPackage/MyCustomLoginView.js
+	 * @attr custom-view
+	 */
+	@property({ attribute: 'custom-view' })
+	customView?: string;
+
+	/**
+	 * Gets or sets the display name of the provider.
+	 *
+	 * @attr display-name
+	 * @example Google
+	 */
+	@property({ attribute: 'display-name' })
+	displayName = '';
+
+	/**
+	 * Gets or sets the name of the provider (otherwise known as authentication type).
+	 *
+	 * @attr provider-name
+	 * @example Umbraco.Google
+	 */
+	@property({ attribute: 'provider-name' })
+	providerName = '';
+
+	/**
+	 * Gets or sets the url to the external login provider.
+	 *
+	 * @attr external-login-url
+	 * @example /umbraco/ExternalLogin
+	 */
+	@property({ attribute: 'external-login-url' })
+	get externalLoginUrl() {
+		return this.#externalLoginUrl;
+	}
+	set externalLoginUrl(value: string) {
+		const tempUrl = new URL(value, window.location.origin);
+		const searchParams = new URLSearchParams(tempUrl.search);
+		tempUrl.searchParams.append('redirectUrl', decodeURIComponent(searchParams.get('returnPath') ?? ''));
+		this.#externalLoginUrl = tempUrl.pathname + tempUrl.search;
+	}
+
+	/**
+	 * Gets or sets the icon to display next to the provider name.
+	 * This should be the name of an icon in the Umbraco Backoffice icon set.
+	 *
+	 * @attr icon
+	 * @example icon-google-fill
+	 * @default icon-lock
+	 */
+	@property({ attribute: 'icon' })
+	icon = 'icon-lock';
+
+	/**
+	 * Gets or sets the look of the underlying uui-button.
+	 *
+	 * @attr button-look
+	 * @example outline
+	 * @default outline
+	 * @see https://uui.umbraco.com/?path=/story/uui-button--looks-and-colors
+	 */
+	@property({ attribute: 'button-look' })
+	buttonLook: InterfaceLook = 'outline';
+
+	/**
+	 * Gets or sets the color of the underlying uui-button.
+	 *
+	 * @attr button-color
+	 * @example danger
+	 * @default default
+	 * @see https://uui.umbraco.com/?path=/story/uui-button--looks-and-colors
+	 */
+	@property({ attribute: 'button-color' })
+	buttonColor: InterfaceColor = 'default';
+
+	#externalLoginUrl = '';
+
+	protected render() {
+		return this.customView
+			? until(this.renderCustomView(), html`<uui-loader-bar></uui-loader-bar>`)
+			: this.renderDefaultView();
+	}
+
+	protected renderDefaultView() {
+		return html`
+			<form id="defaultView" method="post" action=${this.externalLoginUrl}>
+				<uui-button
+					type="submit"
+					name="provider"
+					.value=${this.providerName}
+					title=${`Login using your ${this.displayName} account`}
+					.label=${until(umbLocalizationContext.localize('login_signInWith', undefined, 'Sign in with')) + ' ' + this.displayName}
+					.look=${this.buttonLook}
+					.color=${this.buttonColor}>
+          ${this.displayName
+			  ? html`
+				  <div>
+					  <uui-icon name=${this.icon} aria-hidden="true"></uui-icon>
```

Comment:
@iOvergaard we can remove `aria-hidden` we v1.5.0 of UI library right? 😊

Reply:
Yes, we can thanks to you! But I'm not sure UUI 1.5.0 will be released before Umbraco 13 RC.

Reply:
Okay, we can change it later then 🙂
Btw. did you see this comment?
https://github.com/umbraco/Umbraco.UI/pull/536#issuecomment-1766261377